dnf/base.py

```diff
@@ -984,16 +984,11 @@ def _run_transaction(self, cb):
         if errors is None:
             pass
         elif len(errors) == 0:
-            # this is a particularly tricky case happening also when rpm failed
-            # to obtain the transaction lock. We can only try to see if a
-            # particular element failed and if not, decide that is the
-            # case.
             failed = [el for el in self._ts if el.Failed()]
             if not failed:
-                login = dnf.util.get_effective_login()
-                msg = _("Failed to obtain the transaction lock "
-                        "(logged in as: %s).")
-                logger.critical(msg, login)
+                # if there is no failing element just pass the rpm logs on
+                for msg in cb.messages():
+                    logger.critical(_('RPM: {}').format(msg))
```

Comment:
I could probably get to the bottom myself, but...

The code is hard to read. So if there are no failed elements, you log errors, if there are failed elements, you do nothing? The failed elements are never used, doesn't make much sense... Perhaps worth more explanatory comments?

Reply:
I tried to formulate more explanatory comment.